stl/inc/functional

```diff
@@ -1086,6 +1095,14 @@ public:
 
     template <class _Fx, typename _Mybase::template _Enable_if_callable_t<_Fx, function> = 0>
     function(_Fx&& _Func) {
+        static_assert(is_copy_constructible_v<decay_t<_Fx>>,
+            "The target function object type must be copy constructible (N4988 [func.wrap.func.con]/10.1).");
+        this->_Reset(_STD forward<_Fx>(_Func));
+    }
+
+    template <class _SecretTag, class _Fx, typename _Mybase::template _Enable_if_callable_t<_Fx, function> = 0,
+        enable_if_t<is_same_v<_SecretTag, _Secret_copyability_ignoring_tag>, int> = 0>
+    explicit function(_SecretTag, _Fx&& _Func) { // used exclusively for packaged_task

```

Comment:
Why is `_SecretTag` a template parameter? Could it be just a parameter of type `_Secret_copyability_ignoring_tag` ?

Reply:
I guess this is necessary for defending against some convertible-to-almost-all types. But perhaps we don't need it as of C++17 since the `allocator_arg_t` ctors are removed.

Reply:
The `_SecretTag` should probably be first so we avoid evaluating the other constraint when it's not satisfied.

Reply:
Constraints reordered.